### PR TITLE
Add optimization in openssl detection logic to consume less memory

### PR DESCRIPTION
**Pull request by h0x0er**
Refer #439

### CODE REVIEW

File: user/module/probe_openssl_lib.go [hunk 1]
```diff
@@ -119,67 +118,63 @@ func (m *MOpenSSLProbe) detectOpenssl(soPath string) error {
...
-	var slice [][]byte
-	if slice = bytes.Split(buf, []byte("\x00")); slice == nil {
 		return nil
 	}
 
-	dumpStrings := make(map[uint64][]byte, len(slice))
-	length := uint64(len(slice))
+	buf := make([]byte, 1024*1024) // 1Mb
+	totalReadCount := 0
+	for totalReadCount < int(sectionSize) {
```
Comment (cfc4n) [inline user/module/probe_openssl_lib.go:146]: If the string "OpenSSL 1.1.1j 16 Feb 2021" is split into two buffers, will it be missed?

File: user/module/probe_openssl_lib.go [hunk 2]
```diff
@@ -119,67 +118,63 @@ func (m *MOpenSSLProbe) detectOpenssl(soPath string) error {
...
-		dumpStrings[offset] = slice[i]
+		match := rex.Find(buf)
+		if match != nil {
+			versionKey = string(match)
+			break
+		}
 
-		offset += (uint64(len(slice[i])) + 1)
-	}
+		totalReadCount += readCount
```
Comment (cfc4n) [inline user/module/probe_openssl_lib.go:164]: Can we make some retrospective adjustments here? For example, `totalReadCount += readCount-OpenSslVersionLen`? 

Move forward by an offset of length `OpenSslVersionLen` to address the previously mentioned issue? 

Additionally, `const OpenSslVersionLen = 30 // openssl version string length`.

Comment (h0x0er) [reply user/module/probe_openssl_lib.go:164]: Nice observation @cfc4n , using this adjustment we would be able to cover the edge case. :tada:  Thanks !

**Review by cfc4n — APPROVED** [review]
LGTM,thanks.

**Pull request merged**